### DISCUSSION TITLE
query integer in a column of array of object

### DISCUSSION THREAD

**Discussion by hmbenhaim** [supabase/supabase#8021] [Questions]
Hi
i have a column of array of json and it look like something like that  
...
items: [{
id: "1",
price: 5,
}]
...
how do i query gte or lte on the price field in the array of objects
i could not find any way to do so i saw only way to find contain query 
thank you

**Comment by steve-chavez** [comment]
Assuming your column is `items`, try:

```js
const { data, error } = await supabase
  .from('tbl')
  .select()
  .lte('items->0->>price', 6)
```

https://supabase.com/docs/reference/javascript/select#querying-json-data

**Reply by hmbenhaim** [reply]
Hi 
Thank you for the reply 
if i understand correctly the 0 in 'items->0->>price' is to query the first cell of the array,
How do i query any of the matchings in the array,
i understand that you can't get only the items above 6 so if there is any less than 6 it will mark it as false
what i mean i want all the rows that at least one of their items the price is 6 or above.

**Comment by nilooy** [comment]
## Here is a way to do with database function,

> ####  just to be sure about the data structure, I'm including table and data creation

### Create table
```sql
create table example(
id bigint generated by default as identity primary key, 
items jsonb);
```

### Insert data
```sql
insert into example
  (items)
values
  (
    '[{"id": 1, "price": 6}]'
  ),
  (
     '[{"id": 2, "price": 3}]'
  );
```

### Create a Database function in `supabase studio`
in the query, we will use jsonb_array_elements as the function to structure our data and after that, we can access it directly to filter.

```sql
create or replace function get_item_with_price (min_price decimal)
  returns setof example
  language plpgsql
  as
  $$
    begin
    return query
      select id, items
      from example
      cross join jsonb_array_elements(items) as sub
      where (sub->>'price')::decimal > min_price;
    end;
  $$
```

### Call with supabase js SDK
```js
let { data, error } = await supabase.rpc('get_item_with_price', {min_price: 5})

console.log({data, error})
```


Hope that helps!!
